activerecord/test/cases/associations/has_many_through_associations_test.rb

```diff
@@ -1183,7 +1187,7 @@ def test_save_bang_should_raise_exception_when_join_record_has_errors
       Categorization.validate { |r| r.errors[:base] << "Invalid Categorization" }
       c = Category.new(name: "Fishing", authors: [Author.first])
       assert_raises(ActiveRecord::RecordInvalid) do
-        c.save!
+        assert_deprecated { c.save! }
```

Comment:
I'm curious why the `c.save!` raises a deprecation message.

Reply:
The line 1187 used `[] <<` in the validator. So during save, that validator will be triggered and raise deprecation.